### PR TITLE
Small code style change

### PR DESCRIPTION
**Pull request by cpsw**
A code style suggestion.

### CODE REVIEW

**Comment by miri64** [comment]
Welcome to RIOT @cpsw. Please prefix the commit with the module you are changing. Also, can you please provide a justification for your PR?

**Comment by smlng** [comment]
We appreciate contributions on enhancing RIOTs code in style, readability and documentation, but the proposed changes are (IMHO) rather subjective to personal coding style. In this case the ternary operator is perfectly valid and used properly. In general: using the ternary operator is not against RIOTs coding conventions. We had discussion on that several times and - for sure - there are places where the ternary operator is not appropriate or falsely applied, however, this is not the case here.

I'm against these changes, otherwise we'll likely have lots of such changes back and forth.

**Comment by miri64** [comment]
Hi @smlng, in general, I agree with you, but I also would like @cpsw defend their contribution.

**Comment by smlng** [comment]
Sure! No need to rush, just wanted to raise my concerns.

**Comment by kaspar030** [comment]
> In general: using the ternary operator is not against RIOTs coding conventions. We had discussion on that several times and - for sure - there are places where the ternary operator is not appropriate or falsely applied, however, this is not the case here.

+1

**Comment by cpsw** [comment]
Hi @miri64, thanks! I'm a professor, teaching programming for the last 5 years. I'm concerned with code understanding. I know that if ternary is something really common in practice. However, for beginners, someone that has never seen that, it requires an additional knowledge to understanding to code. Furthermore, in my opinion, it is clearer and faster to see the alternatives when we have an if/else statement, as I suggest. 

@kaspar030, can you please point me to places where if ternary operator is not appropriate or falsely applied? Thanks

**Comment by miri64** [comment]
> @kaspar030, can you please point me to places where if ternary operator is not appropriate or falsely applied? Thanks

@cpsw it was never merged, but see e.g. this discussion: https://github.com/RIOT-OS/RIOT/pull/7276#discussion_r125044201

**Comment by kaspar030** [comment]
Ternary operators are fine. Don't take the one precious little syntactically hygenic line-saver from us poor low-level C coders! ;)

Closing as wontfix.

File: core/clist.c [hunk 1]
```diff
@@ -77,7 +77,11 @@ clist_node_t *_clist_sort(clist_node_t *list, clist_cmp_func_t cmp)
             psize = 0;
             for (i = 0; i < insize; i++) {
                 psize++;
-                q = (q->next == oldhead) ? NULL : q->next;
```
[the remark this answers is not in the document]
Comment (smlng) [reply core/clist.c]: mhm nice idea to keep the 80 chars limit ... in same cases.

**Pull request closed without merging**